mmtrack/models/backbones/mixformer_backbone.py

```diff
@@ -0,0 +1,733 @@
+
+        x = self.proj(x)
+        x = self.proj_drop(x)
+
+        return x
+
+
+class Block(nn.Module):
+
+    def __init__(self,
+                 dim_in,
+                 dim_out,
+                 num_heads,
+                 mlp_ratio=4.,
+                 qkv_bias=False,
+                 drop=0.,
+                 attn_drop=0.,
+                 drop_path=0.,
+                 act_layer=nn.GELU,
+                 norm_layer=nn.LayerNorm,
+                 freeze_bn=False,
+                 **kwargs):
+        super().__init__()
+
+        self.with_cls_token = kwargs['with_cls_token']
+
+        self.norm1 = norm_layer(dim_in)
+        self.attn = Attention(
+            dim_in,
+            dim_out,
+            num_heads,
+            qkv_bias,
+            attn_drop,
+            drop,
+            freeze_bn=freeze_bn,
+            **kwargs)
+
+        self.drop_path = DropPath(drop_path) \
+            if drop_path > 0. else nn.Identity()
+        self.norm2 = norm_layer(dim_out)
+
+        dim_mlp_hidden = int(dim_out * mlp_ratio)
+        self.mlp = Mlp(
+            in_features=dim_out,
+            hidden_features=dim_mlp_hidden,
+            act_layer=act_layer,
+            drop=drop)
+
+    def forward(self, x, t_h, t_w, s_h, s_w):
+        res = x
+        x = self.norm1(x)
+        attn = self.attn(x, t_h, t_w, s_h, s_w)
+        x = res + self.drop_path(attn)
+        x = x + self.drop_path(self.mlp(self.norm2(x)))
+
+        return x
+
+    def forward_test(self, x, s_h, s_w):
+        res = x
+
+        x = self.norm1(x)
+        attn = self.attn.forward_test(x, s_h, s_w)
+        x = res + self.drop_path(attn)
+        x = x + self.drop_path(self.mlp(self.norm2(x)))
+        return x
+
+    def set_online(self, x, t_h, t_w):
+        res = x
+        x = self.norm1(x)
+        attn = self.attn.set_online(x, t_h, t_w)
+        x = res + self.drop_path(attn)
+        x = x + self.drop_path(self.mlp(self.norm2(x)))
+        return x
+
+
+class ConvEmbed(nn.Module):
+    """Image to Conv Embedding."""
+
+    def __init__(self,
+                 patch_size=7,
+                 in_chans=3,
+                 embed_dim=64,
+                 stride=4,
+                 padding=2,
+                 norm_layer=None):
+        super().__init__()
+        patch_size = to_2tuple(patch_size)
+        self.patch_size = patch_size
+
+        self.proj = nn.Conv2d(
+            in_chans,
+            embed_dim,
+            kernel_size=patch_size,
+            stride=stride,
+            padding=padding)
+        self.norm = norm_layer(embed_dim) if norm_layer else None
+
+    def forward(self, x):
+        x = self.proj(x)
+
+        B, C, H, W = x.shape
+        x = rearrange(x, 'b c h w -> b (h w) c').contiguous()
+        if self.norm:
+            x = self.norm(x)
+        x = rearrange(x, 'b (h w) c -> b c h w', h=H, w=W).contiguous()
+
+        return x
+
+
+class VisionTransformer(BaseModule):
```

Comment:
Please check whether we can use the existing module in this file: https://github.com/open-mmlab/mmcv/blob/45fa3e44a22cd0b74071237304bdfef97b5b9bd6/mmcv/cnn/bricks/transformer.py#L632. For examples, `TransformerLayerSequence` , `BaseTransformerLayer`

Reply:
It seems hard to reuse these modules in mmcv, because our backbone reuse the [official implementation of CVT](https://github.com/MCG-NJU/MixFormer) which is considerably different from standard transformer. For example, it apply convolution for projecting to qkv tokens.
And we modify it for sot task, specifically it takes in two images and contains tricky interacting options like mixed attention proposed in our paper.